builder/deploy/uploader.ts

```diff
@@ -9,60 +9,92 @@ import {
   getAccessKeyId,
   getBucket,
   getRegion,
-  getSecretAccessKey,
+  getSecretAccessKey
 } from './config';
 
 export class Uploader {
   private _context: BuilderContext;
+  private _s3: AWS.S3;
+  private _bucket: string;
+  private _region: string;
+  private _builderConfig: Schema;
 
-  constructor(context: BuilderContext) {
+  constructor(context: BuilderContext, builderConfig: Schema) {
     this._context = context;
+    this._builderConfig = builderConfig;
+    this._bucket = getBucket(this._builderConfig);
+    this._region = getRegion(this._builderConfig);
+
+    AWS.config.update({ region: this._region });
+    this._s3 = new AWS.S3({
+      apiVersion: 'latest',
+      secretAccessKey: getSecretAccessKey(this._builderConfig),
+      accessKeyId: getAccessKeyId(this._builderConfig),
+    });
+
+
   }
-  upload(files: string[], filesPath: string, builderConfig: Schema) {
+
+  upload(files: string[], filesPath: string) {
     try {
-      const bucket = getBucket(builderConfig);
-      const region = getRegion(builderConfig);
-      if (!region || !bucket) {
+
+      if (!this._region || !this._bucket) {
         this._context.logger.error(
           `❌  Looks like you are missing some configuration`,
         );
         return;
       }
+
+      const params: HeadBucketRequest = {
+        Bucket: this._bucket,
+      };
+
+      this._s3.headBucket(params)
+        .promise()
+        .then(() => {
+          return this.uploadFiles(files, filesPath);
+        })
+        .catch(() => {
+          this._context.logger.error(
+            `❌  Looks like your credentials are incorrect`,
+          );
```

Comment:
Could we log the error here. I think something like the following will be enough:
```
        .catch(error => {
          this._context.logger.error(`❌  The following error was found during the upload ${error}`);
```